homeassistant/components/wallbox/__init__.py

```diff
@@ -122,20 +122,33 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         hass,
     )
 
-    await wallbox_coordinator.async_validate_input()
+    try:
+        await wallbox_coordinator.async_validate_input()
 
-    await wallbox_coordinator.async_config_entry_first_refresh()
+    except InvalidAuth:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH, "entry_id": entry.entry_id},
+                data=entry.data,
+            )
+        )
+        return False
 
-    hass.data.setdefault(DOMAIN, {CONF_CONNECTIONS: {}})
-    hass.data[DOMAIN][CONF_CONNECTIONS][entry.entry_id] = wallbox_coordinator
```

Comment:
Since you are only storing the coordinator, drop `CONF_CONNECTIONS` and store it as 

`hass.data[DOMAIN][entry.entry_id] = wallbox_coordinator`

Reply:
I changed this in a new commit, did however need the `  hass.data.setdefault(DOMAIN, {entry.entry_id: {}})` part, otherwise it would result in an error